### ISSUE TITLE
GdScript related crash when loading game

### ISSUE THREAD

**Issue by mbrlabs** [godotengine/godot#44238]
**Godot version:**
3.2 666e8ca38779949f2a7f805b913482073bcd5a50

**OS/device including version:**
Windows 10

**Issue description:**
I tried to open my project using the latest `3.2` build and it crashed (works in 3.2.4 beta3). I couldn't reproduce it in a MRP and i don't want to share my project so i looked at open source games/demos and [this demo](https://github.com/GDQuest/godot-2d-jrpg-combat) by GDQuest has exactly the same issue.

I tracked it down to this PR #44093.
CC @pouleyKetchoupp

```
[0] GDScriptParser::_is_type_compatible (C:\Users\mbr\Documents\Projects\godot\modules\gdscript\gdscript_parser.cpp:6296)
[1] GDScriptParser::_reduce_function_call_type (C:\Users\mbr\Documents\Projects\godot\modules\gdscript\gdscript_parser.cpp:7414)
[2] GDScriptParser::_reduce_node_type (C:\Users\mbr\Documents\Projects\godot\modules\gdscript\gdscript_parser.cpp:6504)
[3] GDScriptParser::_check_block_types (C:\Users\mbr\Documents\Projects\godot\modules\gdscript\gdscript_parser.cpp:8286)
[4] GDScriptParser::_check_class_blocks_types (C:\Users\mbr\Documents\Projects\godot\modules\gdscript\gdscript_parser.cpp:8212)
[5] GDScriptParser::_parse (C:\Users\mbr\Documents\Projects\godot\modules\gdscript\gdscript_parser.cpp:8755)
[6] GDScriptParser::parse (C:\Users\mbr\Documents\Projects\godot\modules\gdscript\gdscript_parser.cpp:8827)
[7] GDScript::reload (C:\Users\mbr\Documents\Projects\godot\modules\gdscript\gdscript.cpp:594)
[8] ResourceFormatLoaderGDScript::load (C:\Users\mbr\Documents\Projects\godot\modules\gdscript\gdscript.cpp:2322)
[9] ResourceLoader::_load (C:\Users\mbr\Documents\Projects\godot\core\io\resource_loader.cpp:270)
[10] ResourceLoader::load (C:\Users\mbr\Documents\Projects\godot\core\io\resource_loader.cpp:402)
[11] ResourceInteractiveLoaderText::poll (C:\Users\mbr\Documents\Projects\godot\scene\resources\resource_format_text.cpp:433)
[12] ResourceFormatLoader::load (C:\Users\mbr\Documents\Projects\godot\core\io\resource_loader.cpp:199)
[13] ResourceLoader::_load (C:\Users\mbr\Documents\Projects\godot\core\io\resource_loader.cpp:270)
[14] ResourceLoader::load (C:\Users\mbr\Documents\Projects\godot\core\io\resource_loader.cpp:402)
[15] EditorNode::load_scene (C:\Users\mbr\Documents\Projects\godot\editor\editor_node.cpp:3482)
[16] EditorNode::_sources_changed (C:\Users\mbr\Documents\Projects\godot\editor\editor_node.cpp:807)
[17] MethodBind1<EditorNode,bool>::call (C:\Users\mbr\Documents\Projects\godot\core\method_bind.gen.inc:867)
[18] Object::call (C:\Users\mbr\Documents\Projects\godot\core\object.cpp:919)
[19] Object::emit_signal (C:\Users\mbr\Documents\Projects\godot\core\object.cpp:1246)
[20] Object::emit_signal (C:\Users\mbr\Documents\Projects\godot\core\object.cpp:1304)
[21] EditorFileSystem::_notification (C:\Users\mbr\Documents\Projects\godot\editor\editor_file_system.cpp:1176)
[22] EditorFileSystem::_notificationv (C:\Users\mbr\Documents\Projects\godot\editor\editor_file_system.h:108)
[23] Object::notification (C:\Users\mbr\Documents\Projects\godot\core\object.cpp:931)
[24] SceneTree::_notify_group_pause (C:\Users\mbr\Documents\Projects\godot\scene\main\scene_tree.cpp:974)
[25] SceneTree::idle (C:\Users\mbr\Documents\Projects\godot\scene\main\scene_tree.cpp:525)
[26] Main::iteration (C:\Users\mbr\Documents\Projects\godot\main\main.cpp:2117)
[27] OS_Windows::run (C:\Users\mbr\Documents\Projects\godot\platform\windows\os_windows.cpp:3451)
[28] widechar_main (C:\Users\mbr\Documents\Projects\godot\platform\windows\godot_windows.cpp:162)
[29] _main (C:\Users\mbr\Documents\Projects\godot\platform\windows\godot_windows.cpp:186)
[30] main (C:\Users\mbr\Documents\Projects\godot\platform\windows\godot_windows.cpp:196)
[31] __scrt_common_main_seh (d:\agent\_work\63\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[32] BaseThreadInitThunk
-- END OF BACKTRACE --
```

**Steps to reproduce:**
Just open the project.

**Comment by avencherus** [comment]
I'm on Win10, and wasn't able to reproduce it unfortunately.  Used beta3 and a few other version.  Is there something more maybe?  Wonder what GDScript its loading.


![image](https://user-images.githubusercontent.com/13004169/101701693-122f8700-3a88-11eb-8c17-6f7d92420cab.png)

**Comment by mbrlabs** [comment]
@avencherus This was only recently introduced and is not in `3.2.4 beta3`. You need to build the current `3.2` branch from source to reproduce this.

**Comment by pouleyKetchoupp** [comment]
@mbrlabs Thanks for catching this! I can confirm the regression from #44093 and I'm going to make a fix soon.

**Comment by akien-mga** [comment]
Fixed by #44245.
